### PR TITLE
some additions + changes

### PR DESCRIPTION
**Pull request by sebastienpattyn93**
## Added

- README.md
- wheelhouse.txt (for jujubigdata.utils)
- ES icon.svg

#### config.yaml
- added firewall_enabled option
  - this is default set to false since every charm that gets deployed has it turned off
- added clustername:
  - This will be used later to set the clustername in elasticsearch.yml

#### layer.yaml
- added elasticsearch interface
  - the one in the juju-solutions is not updated yet. [my fork ](https://github.com/sebastienpattyn93/interface-elasticsearch)is up to date and works with this charm

#### metadata.yaml 
- added a provides elasticsearch interface

#### lib/charms/layer/elasticsearch_base.py
- removed unused imports
- counter kept going even when it was >100 because charm stayed in de nested while where request was != 200 => loop.  Combined them into 1 check

#### reactive/elasticsearch_base.py 
- cleaned up the imports with pylint
- LXD problem => environment was not set in the current session so setting it in `/etc/environment` and exporting it afterwards makes it available in the current session
- added some FW methods to add or delete rules to UFW
- first steps for the client relation
- added en changed some events for more detailed steps when configuration will be implemented

### CODE REVIEW

**Comment by sebastienpattyn93** [comment]
To test the relationship, I'm currently working on a xenial metricbeat charm and then add their relation. Still WIP but you can find it [here](https://github.com/sebastienpattyn93/Metricbeat-charm)

**Comment by sebastienpattyn93** [comment]
I discovered that the code to set the environment variable will not work in this charm. Even when I try to export it is still not installing appearantly. It does work in debug-hooks since that is a new session. Can it be a possibility to write thet LXD check in another layer and build ES base layer on top of that? I think that might solve the problem

**Comment by jamesbeedy** [comment]
@sebastienpattyn93 what do you mean by work?

File: lib/charms/layer/elasticsearch_base.py [hunk 1]
```diff
@@ -34,19 +35,18 @@ def es_version():
...
-                    log("Polling for elasticsearch api: %d" % counter)
-                    req = requests.get('http://localhost:9200')
-                    status_code = req.status_code
-                    es_curl_data = req.text
-                    es_vers_str = es_curl_data.strip()
-                    json_acceptable_data = es_vers_str.replace("\n","").replace("'","\"")
-                    return json.loads(json_acceptable_data)['version']['number']
-                except requests.exceptions.ConnectionError:
-                    sleep(1)
+        while status_code != 200 and counter < 100:
```
Comment (jamesbeedy) [inline lib/charms/layer/elasticsearch_base.py]: Nice!

**Comment by jamesbeedy** [comment]
@sebastienpattyn93 these are great adds. Thanks! +1

**Pull request merged**